File: python/program/Car.py
from Vehicle import Vehicle

class Car(Vehicle):
    def __init__(self, plat, merk, tahun, warna, j_kursi, j_pintu):
        super().__init__(plat, merk, tahun, warna)
        self.__j_kursi = j_kursi
        self.__j_pintu = j_pintu

    def get_j_kursi(self):
        return self.__j_kursi

    def set_j_kursi(self, j_kursi):
        self.__j_kursi = j_kursi

    def get_j_pintu(self):
        return self.__j_pintu

    def set_j_pintu(self, j_pintu):
        self.__j_pintu = j_pintu

File: cpp/program/Car.cpp


#include <bits/stdc++.h>
#include "Vehicle.h"

using namespace std;

class Car : public Vehicle
{
private:
    int j_kursi;
    int j_pintu;

public:
    Car()
    {

    }

    Car(string plat, string merk, int tahun, string warna, int j_kursi, int j_pintu)
        : Vehicle(plat, merk, tahun, warna), j_kursi(j_kursi), j_pintu(j_pintu)
    {

    }

    void setJumlahKursi(int j_kursi)
    {
        this->j_kursi = j_kursi;
    }

    int getJumlahKursi() const
    {
        return j_kursi;
    }

    void setJumlahPintu(int j_pintu)
    {
        this->j_pintu = j_pintu;
    }

    int getJumlahPintu() const
    {
        return j_pintu;
    }

    ~Car()
    {

    }
};
